website/docs/r/secrets_sync_aws_destination.html.md

```diff
@@ -0,0 +1,74 @@
+---
+layout: "vault"
+page_title: "Vault: vault_secrets_sync_aws_destination resource"
+sidebar_current: "docs-vault-resource-secrets-sync-aws-destination"
+description: |-
+  Creates an AWS destination to synchronize secrets in Vault
+---
+
+# vault\_secrets\_sync\_aws\_destination
+
+Creates an AWS destination to synchronize secrets in Vault. Requires Vault 1.16+.
+*Available only for Vault Enterprise*.
+
+~> **Important** All data provided in the resource configuration will be
+written in cleartext to state and plan files generated by Terraform, and
+will appear in the console output when Terraform runs. Protect these
+artifacts accordingly. See
+[the main provider documentation](../index.html)
+for more details.
+
+For more information on syncing secrets with AWS, please refer to the Vault
+[documentation](https://developer.hashicorp.com/vault/docs/sync/awssm).
+
+## Example Usage
+
+```hcl
+resource "vault_secrets_sync_aws_destination" "aws" {
+  name              = "aws-dest"
+  access_key_id     = var.access_key_id
+  secret_access_key = var.secret_access_key
+  region            = "us-east-1"
+}
+```
```

Comment:
Should we give an example for the custom_tags & secret_name_template on each destination where applicable? Consistency is always useful, we should decide to either use all optional customisable fields or none in the example usage for all destinations.

Reply:
Added customizable fields to the documentation for all destinations